ipython/Enwiki Revert Classifier.ipynb

```diff
@@ -0,0 +1,301 @@
+      "        <features>            The ClassPath to a list of features to extract.\n",
+      "        --api=<url>           The url of the API to use to extract features\n",
+      "        --language=<clspath>  The ClassPath to a language to use (required for some\n",
+      "                              features)\n",
+      "        --rev_pages=<path>    The location of a file containing rev_ids and\n",
+      "                              page_ids to extract. [default: <stdin>]\n",
+      "\n",
+      "I've already prepared a list of features to extract in ores.models.enwiki.features, so that can just be used by the script. "
+     ]
+    },
+    {
+     "cell_type": "code",
+     "collapsed": false,
+     "input": [
+      "import sys;sys.path.insert(0, \"../\") # Makes ores package accessible\n",
+      "from ores.models.enwiki import features\n",
+      "features"
+     ],
+     "language": "python",
+     "metadata": {},
+     "outputs": [
+      {
+       "metadata": {},
+       "output_type": "pyout",
+       "prompt_number": 1,
+       "text": [
+        "[<log(added_badwords_ratio + 1)>,\n",
+        " <log(added_misspellings_ratio + 1)>,\n",
+        " <log(longest_repeated_char_added + 1)>,\n",
+        " <log(longest_token_added + 1)>,\n",
+        " <log(numeric_chars_added + 1)>,\n",
+        " <log(prev_words + 1)>,\n",
+        " <log(proportion_of_markup_added + 1)>,\n",
+        " <log(proportion_of_numeric_added + 1)>,\n",
+        " <log(proportion_of_symbolic_added + 1)>,\n",
+        " <log(proportion_of_uppercase_added + 1)>,\n",
+        " <log(seconds_since_last_page_edit + 1)>,\n",
+        " <log(words_added + 1)>,\n",
+        " <log(words_removed + 1)>,\n",
+        " <log(user_age_in_seconds + 1)>,\n",
+        " <user_is_anon>,\n",
+        " <user_is_bot>,\n",
+        " <day_of_week_in_utc>,\n",
+        " <hour_of_day_in_utc>,\n",
+        " <is_custom_comment>,\n",
+        " <is_content_namespace>,\n",
+        " <is_section_comment>]"
+       ]
+      }
+     ],
+     "prompt_number": 1
+    },
+    {
+     "cell_type": "markdown",
+     "metadata": {},
+     "source": [
+      "OK.  Time to gather features and reverted status.  To do so, I run the following:\n",
+      "\n",
+      "    $ cat datasets/enwiki.rev_pages.5k.tsv | tail -n+2 | \\\n",
+      "      ./features_reverted \\\n",
+      "          --api=https://en.wikipedia.org/w/api.php \\\n",
+      "          --language=revscores.languages.english \\\n",
+      "          ores.models.enwiki.features > \\\n",
+      "      datasets/enwiki.features_reverted.5k.tsv\n",
+      "\n",
+      "A few of the revisions won't be found -- which is fine.  We'll just have to work with the remaining data.  So, let's load it in!"
+     ]
+    },
+    {
+     "cell_type": "code",
+     "collapsed": false,
+     "input": [
+      "feature_scores = []\n",
+      "with open(\"../datasets/enwiki.features_reverted.5k.tsv\") as f:\n",
```

Comment:
Assumes that the current directory is ore/ipython which is required by the way that "ipython notebook" works.